examples/localvideosnapshot/src/helpers.js

```diff
@@ -10,19 +10,30 @@ var Video = require('twilio-video');
 function displayLocalVideo(video) {
   return Video.createLocalVideoTrack().then(function(localTrack) {
     localTrack.attach(video);
+    return localTrack;
   });
 }
 
 /**
  * Take snapshot of the local video from the HTMLVideoElement and render it
  * in the HTMLCanvasElement.
  * @param {HTMLVideoElement} video
- * @param {HTMLCanvasElement} canvas
+ * @param {LocalVideoTrack} localVideoTrack
+ * @param {HTMLCanvasElement | HTMLImageElement} snapshot
  */
-function takeLocalVideoSnapshot(video, canvas) {
-  var context = canvas.getContext('2d');
-  context.clearRect(0, 0, canvas.width, canvas.height);
-  context.drawImage(video, 0, 0, canvas.width, canvas.height);
+function takeLocalVideoSnapshot(video, localVideoTrack, snapshot) {
+  return new Promise(resolve => {
+    if (window.ImageCapture) {
+      const imageCapture = new ImageCapture(localVideoTrack.mediaStreamTrack);
+      imageCapture.takePhoto().then(function(blob) {
+        snapshot.src = URL.createObjectURL(blob);
+        resolve(URL.createObjectURL(blob));
+      });
+    } else {
+      snapshot.getContext('2d').drawImage(video, 0, 0);
+      resolve(snapshot.toDataURL());
+    }
+  });
```

Comment:
We are calling `takeLocalVideoSnapshot` in index.js but we are not really using the return value. Maybe remove the promise here? Something like below.

```js
function takeLocalVideoSnapshot(video, localVideoTrack, snapshot) {
  if (window.ImageCapture) {
    const imageCapture = new ImageCapture(localVideoTrack.mediaStreamTrack);
    imageCapture.takePhoto().then(function(blob) {
      snapshot.src = URL.createObjectURL(blob);
    });
  } else {
    snapshot.getContext('2d').drawImage(video, 0, 0);
  }
}
```